src/core/border_router/routing_manager.cpp

```diff
@@ -817,6 +817,45 @@ bool RoutingManager::NetworkDataContainsUlaRoute(void) const
     return contains;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_REACHABLITY_CHECK_ICMP6_ERROR_ENABLE
+
+void RoutingManager::CheckReachbilityToSendIcmpError(const Message &aMessage, const Ip6::Header &aIp6Header)
+{
+    Ip6::MessageInfo messageInfo;
+
+    VerifyOrExit(IsRunning() && IsInitalPolicyEvaluationDone());
+
+    VerifyOrExit(!aIp6Header.GetDestination().IsMulticast());
+
+    // Validate that source matches the ULA OMR prefix (which is not
+    // infrastructure-derived).
+
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsEmpty());
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsInfrastructureDerived());
+    VerifyOrExit(aIp6Header.GetSource().MatchesPrefix(mOmrPrefixManager.GetFavoredPrefix().GetPrefix()));
+
+    VerifyOrExit(mRoutePublisher.IsPublishingDefaultRoute());
```

Comment:
this check is not required by the spec? Perhaps we can remove it? The spec requirement also holds for packets sent to an IPv6 destination with a bogus non-existent ULA, and routed to the BR via fc00::/7, for example. 
(Or maybe there's a reason to narrow down the spec requirement on this?)

Reply:
Make sense to me to remove this check and apply the check more generally.